### PR TITLE
Setup test workflow for  various OSs

### PR DESCRIPTION
**Pull request by manuGil**
This is an initial workflow for testing the installation instructions in various Os. Later today we will continue with #33

### CODE REVIEW

**Comment by manuGil** [comment]
I added the installations steps from the lessons. However, the windows job keeps failing
@Selkubi could you please keep debugging this workflow?

The windows job fails because of the R version (strange enough, it worked the very first time). The job fails in step:

```bash
Run r-lib/actions/setup-r@v2
D:\a\_temp\R-4.3.3-win.exe /VERYSILENT /SUPPRESSMSGBOXES /DIR=C:\R
Downloading https://cran.rstudio.com/bin/windows/Rtools/Rtools4.3.x.exe...
Error: Failed to get R 4.3.3: Failed to get R 4.3.3: Failed to download version [object Object]: Error: Unexpected HTTP response: 404
```

**Comment by cforgaci** [comment]
@manuGil, nice! That is a 404 response, so the link does not seem to be correct. Does this work instead? https://cran.r-project.org/bin/windows/Rtools/rtools43/files/rtools43-5958-5975.exe

**Comment by manuGil** [comment]
> @manuGil, nice! That is a 404 response, so the link does not seem to be correct. Does this work instead? https://cran.r-project.org/bin/windows/Rtools/rtools43/files/rtools43-5958-5975.exe

@cforgaci The link is being provided by an action from https://github.com/r-lib/actions/tree/v2-branch/setup-r The only part that I changed was the R version the workflow uses. So I guess is an issue with their action because it works for the other OSs... Or it could be that for windows it is different. Maybe the following can give us an insight on what the problem is:
- Check if the job work when using only windows,
- Check if the job works when no version of R specified, or pointed to the latest available (which might be different for different OS)

**Comment by Selkubi** [comment]
@manuGil I have tried in my fork
1) only windows 
2) no version line at all
3) older version
and got the same error
`D:\a\_temp\R-4.3.3-win.exe /VERYSILENT /SUPPRESSMSGBOXES /DIR=C:\R
Downloading https://cran.rstudio.com/bin/windows/Rtools/Rtools4.3.x.exe...
Error: Failed to get R 4.3.3: Failed to get R 4.3.3: Failed to download version [object Object]: Error: Unexpected HTTP response: 404`

**Comment by Selkubi** [comment]
Also, when I directly copy the [matrix testing](https://github.com/r-lib/actions/tree/v2-branch/setup-r) and change the R version the test runs smoothly and when I add the os `os: [ubuntu-latest, windows-latest, macos-latest` then it passes the first part and gives an error with the GDAL, GEOS, PROJ4 [like so](https://github.com/Selkubi/r-geospatial-urban-manuGil/actions/runs/8363094778/job/22895255123)

**Comment by manuGil** [comment]
@cforgaci and @Selkubi Thanks for trying things out. I will go back to it on Friday.

**Comment by manuGil** [comment]
I fixed the issue with Windows. The name for the version of RTools was wrongly written (it should be `43` and not the usual `4.3` as I assumed. 

The issue with MacOS is partially solved. I made progress with most of the errors with `brew link`, but not all. It seems to be a conflict between Python versions. The latest MacOs version in Github is Monterey, not Sonoma and that might have something to do with it (not sure).  
@cforgaci could you check if the instruction to install the packages in step `Install GDAL, GEOS, and PROJ.4 (macOS)` are correct for Monterey?

**Comment by manuGil** [comment]
Somona is now available for runners as [latest version for MacOS](https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/)

**Comment by manuGil** [comment]
> I fixed the issue with Windows. The name for the version of RTools was wrongly written (it should be `43` and not the usual `4.3` as I assumed.
> 
> The issue with MacOS is partially solved. I made progress with most of the errors with `brew link`, but not all. It seems to be a conflict between Python versions. The latest MacOs version in Github is Monterey, not Sonoma and that might have something to do with it (not sure). @cforgaci could you please check if the instruction to install the packages in step `Install GDAL, GEOS, and PROJ.4 (macOS)` are correct for Monterey?
> 
> Trace back: https://github.com/manuGil/r-geospatial-urban/actions/runs/8401072439/job/23009071352

Check if installing XCode solves this issue

**Review by cforgaci — COMMENTED** [review]
Thanks @manuGil for this! I suggested an R script to test sf and tidyverse in my last comment.

File: .github/workflows/workshop-setup.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          brew install gdal
+        shell: bash
+      - name: Install GDAL, GEOS, and PROJ.4 (Ubuntu)
+        if: matrix.os == 'ubuntu-latest'
+        run: | 
+          sudo add-apt-repository ppa:ubuntugis -y
+          sudo apt-get update
+          sudo apt-get install libgdal-dev libgeos-dev libproj-dev -y
+        shell: bash
+        # Include Fedora and Arch?
```
[the remark this answers is not in the document]
Comment (manuGil) [reply .github/workflows/workshop-setup.yml:42]: OK. I will add those in the next revision. I opened this issue #45  for that. For now, I suggest we complete the merge for the current tests.

**Comment by manuGil** [comment]
@cforgaci and @Selkubi, the workflow is working now and the tests provide by Claudiu were included. We can merge this PR.

**Comment by cforgaci** [comment]
Excellent. Thanks, @manuGil! I will merge the PR.

**Pull request merged**